File: java/leetcode/valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        if (num <= 1) { return true; }
        int i = 1;
        while (i <= num / 2) {
            if (i * i == num) { return true; }
            i++;
        }
        return false;
    }
}

File: python/leetcode/valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return True
        i = 1
        while i <= num / 2:
            if i * i == num:
                return True
            i += 1
        return False
